src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        return Math.max(tile != null ? tile.getMicroblockContainer().getPartContainer().getLightValue() : 0,
+                getLightValueDefault(state, world, pos));
+    }
+
+    public int getLightValueDefault(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        return super.getLightValue(state, world, pos);
+    }
+
+    @Override
+    public final ItemStack getPickBlock(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        if (target instanceof RayTraceResult)
+        {
+            IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+            return tile != null ? tile.getMicroblockContainer().getPartContainer().getPickBlock(player, (RayTraceResult) target) : null;
+        }
+        return getPickBlockDefault(state, target, world, pos, player);
+    }
+
+    public ItemStack getPickBlockDefault(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        return super.getPickBlock(state, target, world, pos, player);
+    }
+
+    private IMicroblockContainerTile brokenTile = null;
+    private boolean harvestingWrapper = false;
+
+    @Override
+    public final void harvestBlock(World world, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        brokenTile = te instanceof IMicroblockContainerTile ? (IMicroblockContainerTile) te : null;
+        harvestBlockDefault(world, player, pos, state, te, stack);
+        brokenTile = null;
+    }
+
+    public void harvestBlockDefault(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        super.harvestBlock(worldIn, player, pos, state, te, stack);
+    }
+
+    @Override
+    public final List<ItemStack> getDrops(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        if (harvestingWrapper) { return getDropsDefault(world, pos, state, fortune); }
+        IMicroblockContainerTile brokenTile = getMicroblockTile(world, pos);
+        if (brokenTile == null)
+        {
+            brokenTile = this.brokenTile;
+        }
+        if (brokenTile == null) { return getDropsDefault(world, pos, state, fortune); }
+        List<ItemStack> drops = new ArrayList<ItemStack>();
+        drops.addAll(getDropsDefault(world, pos, state, fortune));
+        return drops;
+    }
+
+    public List<ItemStack> getDropsDefault(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        return super.getDrops(world, pos, state, fortune);
+    }
+
+    @Override
+    public final boolean removedByPlayer(IBlockState state, World world, BlockPos pos, EntityPlayer player, boolean willHarvest)
+    {
+        RayTraceResult hit = reTraceAll(world, pos, player);
+        if (hit instanceof RayTraceResult)
+        {
```

Comment:
Yeah... leftovers from PartMOP... You can see I had to do a ton of typecasting all over the place...